manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -217,4 +225,7 @@ spec:
         configMap:
           name: pilot-jwks-extra-cacerts{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
   {{- end }}
+{{- if .Values.pilot.extraVolumes }}
```

Comment:
maybe this test could be updated to use the new fields: https://github.com/istio/istio/blob/0392abbdbf549f44fa6d093d0029e44c119ae081/tests/integration/security/file_mounted_certs/main_test.go#L140

perhaps outside of scope for this PR, but the gateway ones in the same test could also be using the existing `secretVolumes` options present the gateway chart.

adding those to the test will fail until added to `operator/pkg/apis/istio/v1alpha1/values_types.proto` as well, which is required for fields using istioctl

Reply:
makes sense,  i will do the gateways one in a separate commit to not confuse things.

As for the protobuf changes, should i restrict the proto definition to the limited set of secret volumes or allow for example for configmap volumes or real volumes?
if i go for the broad version that's a lot of protos to add:
https://github.com/kubernetes/api/blob/8fb9658ecf05b1e952883268f63ba699ea1e1ad7/core/v1/generated.proto#L5450

if i go for restricted i wonder if there might be some error if people expect to (ab)use it in certain ways and the best way to show that.

Reply:
I have taken your advice to use the more generic proto types, makes perfect sense in this context :+1: 
I will add the gateway bit right away.

Reply:
Ok all tests passed in commit `[96c2b44](https://github.com/istio/istio/pull/37264/commits/96c2b444d6018225c63ffb8d3e00ea4c9d5d3040)`
added the gateway change , and i think that addresses everything?

Reply:
looks like all tests are passing now :+1: